File: src/main/java/xf/xflp/exception/XFLPException.java
package xf.xflp.exception;

/**
 * Copyright (c) 2012-2022 Holger Schneider
 * All rights reserved.
 *
 * This source code is licensed under the MIT License (MIT) found in the
 * LICENSE file in the root directory of this source tree.
 *
 *
 * @author hschneid
 *
 */
public class XFLPException extends RuntimeException {

    private final XFLPExceptionType type;

    public XFLPException(XFLPExceptionType type, String message) {
        super(message);
        this.type = type;
    }

    public XFLPException(XFLPExceptionType type, String message, Exception e) {
        super(message, e);
        this.type = type;
    }

    public XFLPExceptionType getType() {
        return type;
    }
}

File: src/xflp/exception/xflp_exception.py
"""
Exception classes for XFLP.

Copyright (c) 2012-2025 Holger Schneider
Licensed under the MIT License
"""

from enum import Enum, auto


class XFLPExceptionType(Enum):
    """Types of XFLP exceptions."""

    ILLEGAL_STATE = auto()
    ILLEGAL_ARGUMENT = auto()
    ILLEGAL_INPUT = auto()


class XFLPException(Exception):
    """
    Custom exception for XFLP errors.

    Attributes:
        type: The type of exception
        message: Error message
    """

    def __init__(self, exception_type: XFLPExceptionType, message: str,
                 cause: Exception = None):
        """
        Initialize an XFLP exception.

        Args:
            exception_type: Type of the exception
            message: Error message
            cause: Optional underlying exception
        """
        super().__init__(message)
        self.type = exception_type
        self.__cause__ = cause

    def get_type(self) -> XFLPExceptionType:
        """Get the exception type."""
        return self.type
